### PR TITLE
MLIBZ-1967: Fix Storage API

### PR DESCRIPTION
**Pull request by thomasconner**
#### Description
Replace errors to use `KinveyError`. Expose names as adapters and not as the default export.

### CODE REVIEW

**Comment by thomasconner** [comment]
@georgiwe Please take a look at this when you have a chance.

**Comment by gprodanov** [comment]
Apologies for the delay. Looks good. I only wish we'd have class names match file names at some point in the future :)

**Review by gprodanov — APPROVED** [review]

File: src/middleware/src/storage/index.js [hunk 1]
```diff
@@ -1,22 +1,22 @@
 import { isDefined } from 'kinvey-js-sdk/dist/export';
-import KinveyStorage from 'kinvey-js-sdk/dist/request/src/middleware/src/storage';
-import IndexedDB from './src/indexeddb';
-import WebSQL from './src/websql';
-import { LocalStorage } from './src/webstorage';
+import { Storage as KinveyStorage } from 'kinvey-js-sdk/dist/request/src/middleware/src/storage';
+import { IndexedDBAdapter } from './indexeddb';
```
Comment (tejasranade) [inline src/middleware/src/storage/index.js:3]: @thomasconner none of the files have moved as part of this commit, so how/why did the relative paths change?

Comment (thomasconner) [reply src/middleware/src/storage/index.js:3]: Yes the files did move. Please see the path change designated like this

`src/middleware/src/storage/src/indexeddb.js → src/middleware/src/storage/indexeddb.js`

**Pull request merged**